icu4c/source/i18n/smpdtfmt.cpp

```diff
@@ -979,7 +991,7 @@ SimpleDateFormat::initialize(const Locale& locale,
         //fNumberFormat->setLenient(true); // Java uses a custom DateNumberFormat to format/parse
 
         initNumberFormatters(locale, status);
-        initFastNumberFormatters(status);
+        initSimpleNumberFormatter(locale, status);
```

Comment:
A tricky thing is that it could potentially be e.g. a CurrencyFormatter (though that feels nonsensical), ideally the field would be a `DecimalFormat` field.

what i can do is make it adopt the symbols from the number formatter if it is indeed a decimal formatter

Reply:
Curious what @sffc's thoughts are here

We can always share decimal symbols when constructing. Ought we do the same even when adopting (and just set it to null if it's using a currency/etc formatter instead)

In that case the only time fNumberFormat would actually be used is if you adopt a currency (etc) formatter.

(Do we even need fSharedNumberFormatters then?)

It does seem like a lot of this mechanism would go away if we only used fNumberFormat in the case of adoption, and we can deal with the cost of removing fSharedNumberFormatters

Reply:
I'm pretty sure we use the "fancy" NumberFormat for compatibility with date patterns that use algorithmic numbering systems, like Traditional Han (used in some chinese calendar formatting) and Roman Numeral (used in Hawaiian and maybe others).

There might be room to further clean this up; I prefer to make a small PR to take care of the low hanging fruit (which this one does) and then investigate the feasibility of cleaning up the other formatters.

It would be nice to deduplicate the DecimalFormatSymbols, if that works. However, I'm not immediately sure if a DecimalFormatSymbols for an RBNF works with a SNF.

Reply:
oh yeah that's valid. sounds like it's best considered in a followup